src/process_management/process_manager.rs

```diff
@@ -34,7 +38,7 @@ impl ProcessManager {
      * - Windows: Expects `.exe` to be the extension of the executables
      * - Everything Else: Expects no extension on the executables
      */
-    pub fn new(path: &str) -> Result<ProcessManager, &str> {
+    pub fn new(path: &'static str) -> Result<ProcessManager> {
```

Comment:
Do we want to only use program-lifetime string references, or should we convert this to a String value to make it more flexible?

Reply:
Reason why it is static is preserve the lifetime of the `ProcessManagerError` messages, as it borrows the path.

Reply:
But there is a difference between preserving the lifetime for the error messages and requiring the lifetime to be static. Static lifetimes will basically prevent you from using anything generated at runtime without purposefully leaking it, which is obviously the wrong way to do it.
Can you guarantee that all uses of this will use variables from compile time?

Reply:
When I convert this to use an arbitrary lifetime, compiler complains

```
borrowed data escapes outside of associated function
`dir` escapes the associated function body here

borrowed data escapes outside of associated function
argument requires that `'a` must outlive `'static`
```

So some work needs to be done in order for `ProcessManagerError` to work as intended

Reply:
I switched this from static path reference to a PathBuf.
So basically the version of path that allows by value instead of reference.